sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SecretsMatcherFactory.java

```diff
@@ -32,19 +31,12 @@ private SecretsMatcherFactory() {
   }
 
   public static SecretsMatcher constructSecretsMatcher(Rule rule) {
-    if (rule.getDetection().getMatching() instanceof AuxiliaryPattern) {
-      return constructSecretsMatcher((AuxiliaryPattern) rule.getDetection().getMatching());
+    Matching matching = rule.getDetection().getMatching();
+    if (matching != null) {
+      return new RegexMatcher(matching.getPattern());
     } else {
       return NO_DETECTION_MATCHER;
     }
```

Comment:
```suggestion
    return matching != null ? new RegexMatcher(matching.getPattern()) : NO_DETECTION_MATCHER;
```
Only a suggestion. And it would even be nicer with something like `RegexMatcher.build(matching)` :smile:

Reply:
Is applied. Since i am changing the structure in another commit, the builder has to wait for now